ang/civiawards/dashboard/directives/more-filters-dashboard-action-button.controller.js

```diff
@@ -0,0 +1,171 @@
+(function (angular, _) {
+  var module = angular.module('civiawards');
+
+  module.controller('MoreFiltersDashboardActionButtonController', MoreFiltersDashboardActionButtonController);
+
+  /**
+   * Handles the visibility and click event for the "Add Award" dashboard action button.
+   *
+   * @param {object} $rootScope rootscope object
+   * @param {object} $scope scope object
+   * @param {object} $location the location service
+   * @param {object} getSelect2Value service to fetch select 2 values
+   * @param {object} crmApi the service to fetch civicrm backend
+   * @param {object} dialogService the dialog service
+   * @param {object} ts translation service
+   * @param {object} CaseStatus Case Status service
+   * @param {object} AwardType Award Type service
+   */
+  function MoreFiltersDashboardActionButtonController ($rootScope, $scope, $location, getSelect2Value, crmApi, dialogService, ts, CaseStatus, AwardType) {
+    var model = {
+      statuses: _.map(CaseStatus.getAll(), mapSelectOptions),
+      award_types: _.map(AwardType.getAll(), mapSelectOptions),
+      awardOptions: [
+        { text: ts('My Awards'), id: 'my_awards' },
+        { text: ts('All Awards'), id: 'all_awards' }
+      ],
+      selectedFilters: {
+        awardFilter: 'my_awards',
+        statuses: '',
+        award_types: '',
+        start_date: null,
+        end_date: null
+      },
+      applyFilterAndCloseDialog: applyFilterAndCloseDialog
+    };
+
+    $scope.clickHandler = clickHandler;
+    $scope.isVisible = isVisible;
+    $scope.isNotificationVisible = isNotificationVisible;
+
+    (function init () {
+      applyFilter();
+    }());
+
+    /**
+     * Checks if Notification should be visible
+     *
+     * @returns {boolean} if Notification should be visible
+     */
+    function isNotificationVisible () {
+      return !_.isEqual(model.selectedFilters, {
+        awardFilter: 'my_awards',
+        statuses: '',
+        award_types: '',
+        start_date: null,
+        end_date: null
+      });
+    }
+
+    /**
+     * Redirects the user to the awards creation screen.
+     */
+    function clickHandler () {
+      dialogService.open('MoreFilters', '~/civiawards/dashboard/directives/more-filters-popup.html', model, {
+        autoOpen: false,
+        height: 'auto',
+        width: '350px',
+        title: 'More Filters'
+      });
+    }
+
+    /**
+     *
+     */
+    function applyFilterAndCloseDialog () {
+      applyFilter();
+      dialogService.close('MoreFilters');
+    }
+    /**
+     *
+     */
+    function applyFilter () {
+      processMyAwardsFilter()
+        .then(processAwardTypeFilters)
```

Comment:
I think these two can run at the same time so it should be:

```js
$q.all({
  myAwards: processMyAwardsFilter(),
  awardType: processAwardTypeFilters()
})
  .then(function (results) {
    $rootScope.$broadcast('civicase::dashboard-filters::updated', {
      case_type_id: { IN: results.awardType }
    });
  });
```

But, there are two things to consider:

1 - why do `processMyAwardsFilter` and `processAwardTypeFilters` both return a list of awards ids but we only care about the ones returned by `processAwardTypeFilters` ?
2 - can we try to use `applications` and `awards` instead of `awards` and `awardTypes` to avoid confusion? The former are the proper terms for cases and case types.

Reply:
As discussed not changing it.